generators/openapi-client/files.js

```diff
@@ -177,23 +155,36 @@ function writeFiles() {
                 return;
             }
 
+            this.javaDir = `${constants.SERVER_MAIN_SRC_DIR + this.packageFolder}/`;
             const mainClassFile = `${this.javaDir + this.getMainClassName()}.java`;
 
             this.rewriteFile(
                 mainClassFile,
                 'import org.springframework.core.env.Environment;',
-                'import org.springframework.context.annotation.ComponentScan;'
+                'import org.springframework.context.annotation.ComponentScan;\n' +
+                    'import org.springframework.context.annotation.FilterType;'
             );
 
             const componentScan =
-                `${'@ComponentScan( excludeFilters = {\n    @ComponentScan.Filter('}${this.packageName}` +
-                '.client.ExcludeFromComponentScan.class)\n})';
+                '@ComponentScan( excludeFilters = {\n' +
+                '   @ComponentScan.Filter(type = FilterType.REGEX, ' +
+                `pattern = "${this.packageName}.client.*.ClientConfiguration")\n})`;
             this.rewriteFile(mainClassFile, '@SpringBootApplication', componentScan);
-
-            this.template(
-                'src/main/java/package/client/_ExcludeFromComponentScan.java',
-                `${this.javaDir}/client/ExcludeFromComponentScan.java`
-            );
         }
     };
 }
+
+function removeClientFiles(cliName) {
```

Comment:
There is only 1 use if this, I think this unnecessary.
See below.

Reply:
This have been done because the folders were created with snakeCase pattern. We changed to camelCase pattern but for backward compatibility we have to both remove snake and camel case folders.

its ok for me to put it under a specific function to improve the code meaning while having this under a large method bring to more complexity.

Reply:
Just read your message @avdev4j. I wanted to kept it for the same reason but I've already removed it. The code isn't that complex now with these last modifications.

Let me know if you prefer me to restore the arrow function: https://github.com/jhipster/generator-jhipster/blob/master/generators/openapi-client/files.js#L60

Reply:
I suggested to remove it because of the binding.
As alternative it can be placed at the generator with `_` prefix.

Reply:
Actually files.js is pretty simple.
I think it should be moved to the generator itself.

Reply:
Not sure if it's what you meant, but I moved the lines without wrapping them into a function. I think we can resolve this comment and merge.

https://github.com/ecostanzi/generator-jhipster/blob/fed7b9ff30b90120079cbc2b4fd2f592ec0ecdd6/generators/openapi-client/files.js#L47-L52

Reply:
It was just a suggestion.
Delete files.js and put everything inside the generator, it's easier to work with only 1 file.

For me this is resolved.